app/Platform/Commands/SyncAchievementAuthors.php

```diff
@@ -0,0 +1,99 @@
+
+class SyncAchievementAuthors extends Command
+{
+    protected $signature = 'ra:sync:achievement-authors
+                            {achievementId? : Target achievement ID. Leave empty to sync from all achievements.}';
+    protected $description = "Write all LOGIC authorship data to achievement_authors";
+
+    public function __construct(
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(): void
+    {
+        $achievementId = $this->argument('achievementId');
+
+        if ($achievementId) {
+            $achievement = Achievement::findOrFail($achievementId);
+
+            $this->syncAchievementLogicAuthors($achievement);
+        } else {
+            $distinctAchievementIdsCount = Achievement::count();
+
+            $this->info("\nUpserting logic authorship credit for {$distinctAchievementIdsCount} achievements.");
+
+            $progressBar = $this->output->createProgressBar($distinctAchievementIdsCount);
+            Achievement::query()->chunk(100, function ($achievements) use ($progressBar) {
+                foreach ($achievements as $achievement) {
+                    $this->syncAchievementLogicAuthors($achievement, quiet: true);
+
+                    $progressBar->advance();
+                }
+            });
+            $progressBar->finish();
+
+            $this->info("\nCompleted upserting logic authorship credit for {$distinctAchievementIdsCount} achievements.");
+        }
+    }
+
+    private function syncAchievementLogicAuthors(Achievement $achievement, bool $quiet = false): void
+    {
+        if (!$quiet) {
+            $this->info("\nUpserting logic authorship credit for achievement [{$achievement->id}:{$achievement->title}] ({$achievement->game->title}).");
+        }
+
+        $addedAuthors = 0;
+
+        $achievementAuthor = $achievement->developer()->withTrashed()->first();
+
+        if ($achievementAuthor) {
+            $achievement->upsertAuthorshipCredit(
+                $achievementAuthor,
+                AchievementAuthorTask::LOGIC,
+                backdate: $achievement->DateCreated,
+            );
+
+            $addedAuthors++;
+        }
+
+        $systemComments = $achievement->legacyComments()->automated()->get();
+        foreach ($systemComments as $systemComment) {
+            $payload = $systemComment->Payload;
+
+            // Extract the username from the payload.
+            $username = strtok($payload, ' ');
+
+            if (str_contains($payload, 'logic')) {
+                $user = User::withTrashed()
+                    ->where('User', $username)
+                    ->orWhere('display_name', $username)
+                    ->first();
+
+                if ($user) {
+                    $achievement->upsertAuthorshipCredit(
+                        $user,
+                        AchievementAuthorTask::LOGIC,
+                        backdate: $systemComment->Submitted,
+                    );
+
+                    $addedAuthors++;
```

Comment:
The easiest solution to both of these problems would be to keep an array of authors who have been given credit, don't try to upsert authors that are already in the array, and count the number of entries in the array at the end of the function.

Reply:
Might also just want to track the last credited author. If authorship changes, you might need the full trail:

* Date 1 - UserA created achievement - given logic credit
* Date 2 - UserA patched achievement - logic credit not updated
* Date 3 - UserB patched achievement - given logic credit
* Date 4 - UserA patched achievement - given new logic credit?

Or is that the reason that only the newest date is kept? "The logic was last updated at DateX by UserA". Is this meant to just capture the current state, and not the history? 

We currently report data for every user who has touched it (see http://localhost:64000/manage/achievements/44672). Is it more important to keep the most recent edit, or the oldest per user?

Reply:
I think it's more important to keep the oldest credit per user. For detailed trails/history, the (currently unused) `triggers` table seems like where that should go.

If the same user contributes two unique badges to the same achievement, I think it's likely sufficient that they're getting badge credit -- I don't think they'll need two credit rows, and this may lead to more complexity in building out the player-facing UX later on.

Per your suggestion, in latest I've changed `$addedAuthors` to be an array of credited authors so multiple edits by a single author aren't counted.